graylog2-web-interface/src/views/components/queries/QueryTitleEditModal.test.jsx

```diff
@@ -0,0 +1,79 @@
+// @flow strict
+import * as React from 'react';
+import { render, cleanup, fireEvent, wait } from '@testing-library/react';
+
+import QueryTitleEditModal from './QueryTitleEditModal';
+
+jest.mock('views/stores/QueriesStore', () => ({ QueriesActions: {} }));
+jest.mock('views/stores/ViewStore', () => ({ ViewActions: {} }));
+
+describe('QueryTitleEditModal', () => {
+  afterEach(cleanup);
+  const modalHeadline = 'Edit query title';
+  const openModal = (modalRef, currentTitle = 'CurrentTitle') => {
+    if (modalRef) {
+      modalRef.open(currentTitle);
+    }
+  };
+
+  it('shows after triggering open action', () => {
+    let modalRef;
+    const { queryByText } = render(
+      <QueryTitleEditModal ref={(ref) => { modalRef = ref; }}
+                           onTitleChange={() => Promise.resolve()} />,
+    );
+    // Modal should not be visible initially
+    expect(queryByText(modalHeadline)).toBeNull();
+    openModal(modalRef);
+    // Modal should be visible
+    expect(queryByText(modalHeadline)).not.toBeNull();
+  });
+
+  it('has correct initial input value', () => {
+    let modalRef;
+    const { getByDisplayValue } = render(
+      <QueryTitleEditModal ref={(ref) => { modalRef = ref; }}
+                           onTitleChange={() => Promise.resolve()} />,
```

Comment:
I could not use the typical `{() => {}}` due to flow strict here. Is this the best way to deal with this case?

Reply:
Yes that is the best we have.